### PR TITLE
Support skipping tests that don't actually hold data

### PR DESCRIPTION
**Pull request by joehandzik**
Code originated from Craig Tierney at NVIDIA

Signed-Off-By: Joe Handzik <jhandzik@nvidia.com>

#### Why we need this PR?
*Pick one, remove the rest*
- Refactoring to improve handling of datasets for RN50 Pipeline test

#### What happened in this PR?
 - What solution was applied:
     *[ When test data is nonexistent for whatever reason, the test is skipped, avoiding a failure ]*
 - Affected modules and functionalities:
     *[ RN50 Pipeline test ]*
 - Key points relevant for the review:
     *[ ]*
 - Validation and testing:
     *[ Run internally for performance comparison purposes ]*
 - Documentation (including examples):
     *[ N/A]*

**JIRA TASK**: *[NA]*

### CODE REVIEW

File: dali/test/python/test_RN50_data_pipeline.py [hunk 1]
```diff
@@ -263,6 +264,25 @@ def update(self, val, n=1):
 for pipe_name in test_data.keys():
     data_set_len = len(test_data[pipe_name])
     for i, data_set in enumerate(test_data[pipe_name]):
+
+        skiptest=False
+        for d in data_set:
+            if os.path.isdir(d):
+                continue
+            count=0
```
Comment (JanuszL) [inline dali/test/python/test_RN50_data_pipeline.py:272]:
```suggestion
            count = 0
```
Nitpick

File: dali/test/python/test_RN50_data_pipeline.py [hunk 2]
```diff
@@ -263,6 +264,25 @@ def update(self, val, n=1):
...
+            for f in glob.glob(d):
+                if not os.path.exists(f):
+                    skiptest=True
+                count=count+1
```
Comment (JanuszL) [inline dali/test/python/test_RN50_data_pipeline.py:276]:
```suggestion
                count = count + 1
```

Comment (JanuszL) [inline dali/test/python/test_RN50_data_pipeline.py:276]:
```suggestion
                if os.path.exists(f):
                    count=count+1
```

File: dali/test/python/test_RN50_data_pipeline.py [hunk 3]
```diff
@@ -263,6 +264,25 @@ def update(self, val, n=1):
...
+            if count == 0:
+               skiptest=True
```
Comment (JanuszL) [inline dali/test/python/test_RN50_data_pipeline.py:278]:
```suggestion
            if count == 0:
               skiptest=True
               break
```

File: dali/test/python/test_RN50_data_pipeline.py [hunk 4]
```diff
@@ -263,6 +264,25 @@ def update(self, val, n=1):
...
+
+            if skiptest:
+                break
```
Comment (JanuszL) [inline dali/test/python/test_RN50_data_pipeline.py:281]:
```suggestion
        skip_test = False
        for d in data_set:
            if os.path.isdir(d):
                continue
            count = 0
            for f in glob.glob(d):
                if os.path.exists(f):
                    count = count + 1
            if count == 0:
               skip_test = True
               break
```

File: dali/test/python/test_RN50_data_pipeline.py [hunk 5]
```diff
@@ -263,6 +264,25 @@ def update(self, val, n=1):
...
+        if skiptest:
+            print("Files not found, Skipping test {} {}".format(pipe_name.__name__,data_set))
+            break
```
Comment (JanuszL) [inline dali/test/python/test_RN50_data_pipeline.py:284]:
```suggestion
            continue
```

Comment (JanuszL) [reply dali/test/python/test_RN50_data_pipeline.py:284]: Or you just wan to end the whole test?

**Review by nitish-awasthi — APPROVED** [review]

**Comment by JanuszL** [comment]
Closing for now

**Pull request closed without merging**